File: service/TFC.Application.DTO/EntityDTO/RoutineDTO.cs
namespace TFC.Application.DTO.EntityDTO
{
    public class RoutineDTO
    {
        public long RoutineId { get; set; }
        public string? RoutineName { get; set; }
        public string? RoutineDescription { get; set; }
        public long UserId { get; set; }
        public List<SplitDayDTO> SplitDays { get; set; } = new List<SplitDayDTO>();
    }
}

File: tfc_gym_app/lib/data/models/dto/entitie/routine_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:tfc_gym_app/data/models/dto/entitie/split_day_dto.dart';

part 'routine_dto.g.dart';

@JsonSerializable()
class RoutineDTO {
  final int routineId;
  final String? routineName;
  final String? routineDescription;
  final int userId;
  final List<SplitDayDTO>? splitDays;

  RoutineDTO({
    required this.routineId,
    this.routineName,
    this.routineDescription,
    required this.userId,
    this.splitDays,
  });

  factory RoutineDTO.fromJson(Map<String, dynamic> json) => _$RoutineDTOFromJson(json);
  Map<String, dynamic> toJson() => _$RoutineDTOToJson(this);
}